pkg/build/docker_go.go

```diff
@@ -407,28 +549,36 @@ COPY . /
 
 RUN cd ${PLAN_DIR} \
     && go env -w GOPROXY="${GO_PROXY}" \
-    && GOOS=linux GOARCH=amd64 go build -o /testplan ${BUILD_TAGS} ${TESTPLAN_EXEC_PKG}
+    && GOOS=linux GOARCH=amd64 go build -o ${PLAN_DIR}/testplan.bin ${BUILD_TAGS} ${TESTPLAN_EXEC_PKG}
 
 {{.DockerfileExtensions.PostBuild}}
 
 # Store module dependencies
 RUN cd ${PLAN_DIR} \
   && go list -m all > /testground_dep_list
 
-{{ if not .SkipRuntimeImage }}
 #:::
-#::: RUNTIME CONTAINER
+#::: (OPTIONAL) RUNTIME CONTAINER
 #:::
 
-FROM ${RUNTIME_IMAGE} AS binary
+{{ if not .SkipRuntimeImage }}
+
+FROM ${RUNTIME_IMAGE} AS runtime
+
+# PLAN_DIR is the location containing the plan source inside the build container.
+ENV PLAN_DIR /plan
 
```

Comment:
I am missing a section that loads `testplan` in the root `/` dir when `.SkipRuntimeImage` is `false`. I think that would make the image incomptible.